plugins/colorbutton/plugin.js

```diff
@@ -86,6 +86,7 @@ CKEDITOR.plugins.add( 'colorbutton', {
 		function addButton( name, type, title, order, options ) {
 			var style = new CKEDITOR.style( config[ 'colorButton_' + type + 'Style' ] ),
 				colorBoxId = CKEDITOR.tools.getNextId() + '_colorBox',
+				colorData = {},
```

Comment:
Instead of creating `colorData` variable, I'd think about extracting `onOpen` logic into separate function, `getColorFromSelection( editor )`. This function can be later used inside `clickFn` to pass data about color into dialog. IMO it would be more elegant this way.

Reply:
Although I agree that would be a bit cleaner way of doing this unfortunately I found it too difficult to extract `onopen` logic from its current scope. Therefore I suggest to leave the current solution as it is still not that ugly.